src/main/java/com/faforever/api/clan/ClansController.java

```diff
@@ -0,0 +1,81 @@
+package com.faforever.api.clan;
+
+import com.faforever.api.data.domain.Clan;
+import com.faforever.api.data.domain.Player;
+import com.faforever.api.player.PlayerRepository;
+import com.faforever.api.utils.AuthenticationHelper;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.core.Authentication;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+
+@RestController
+@RequestMapping(path = "/clan")
+public class ClansController {
+
+  private final ClanRepository clans;
+  private final PlayerRepository players;
+
+  @Inject
+  public ClansController(ClanRepository clans, PlayerRepository users) {
+    this.clans = clans;
+    this.players = users;
+  }
+
+  @ApiOperation("Create a clan with correct leader, founder and clan membership")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"),
+      @ApiResponse(code = 400, message = "Bad Request")})
+  @RequestMapping(path = "/create", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+  @Transactional
+  public String createClan(HttpServletRequest request,
+                           @RequestParam(value = "name") String name,
+                           @RequestParam(value = "tag") String tag,
+                           @RequestParam(value = "description", required = false) String description,
+                           Authentication authentication) throws IOException, ClanException {
+    Clan myClan = new Clan();
+    myClan.setName(name);
+    myClan.setTag(tag);
+    myClan.setDescription(description);
+
+    Optional<Player> player = AuthenticationHelper.getPlayer(authentication, players);
+    if (!player.isPresent()) {
+      throw new ClanException("cannot find logged in player");
+    }
+    if (player.get().getClan().size() > 0) {
+      throw new ClanException("player has allready a clan");
```

Comment:
`Player` (uppercase) and `already` (spelling). Log the player name as well.
Does `getClan()` return a list or why does it have a `size()`? If so it should have a different name.